### DISCUSSION TITLE
Order only by matching related models count

### DISCUSSION THREAD

**Discussion by semurgx** [prisma/prisma#17690] [Q&A]
Given code
```
prisma.post.findMany({
   orderBy: { postFavorite: 'asc' },
   postFavorite: { some: { userId: 1337 } },
})
```
produces SQL like
```
SELECT *
FROM post
LEFT JOIN (
  SELECT post_favorite.post_id, COUNT(*) AS orderby_aggregator
  FROM post_favorite
  WHERE 1=1
  GROUP BY post_favorite.post_id
) AS orderby_1_Postfavorite ON (post.id = orderby_1_Postfavorite.post_id)
LEFT JOIN (
  SELECT post_favorite.post_id, COUNT(*) AS _aggr_count_favorite
  FROM post_favorite
  WHERE post_favorite.user_id = $1
  GROUP BY post_favorite.post_id
) AS aggr_selection_0_PostFavorite ON (post.id = aggr_selection_0_PostFavorite.post_id)
WHERE 1=1
ORDER BY COALESCE(orderby_1_Postfavorite.orderby_aggregator, $2) ASC
LIMIT $3 OFFSET $4
```
where first join used for sorting and the second for filtering and I'm wondering is there any possibility to order only by matching related models? Like this
```
SELECT *
FROM post
LEFT JOIN (
  SELECT post_favorite.post_id, COUNT(*) AS orderby_aggregator
  FROM post_favorite
  WHERE post_favorite.user_id = $1
  GROUP BY post_favorite.post_id
) AS orderby_1_Postfavorite ON (post.id = orderby_1_Postfavorite.post_id)
WHERE 1=1
ORDER BY COALESCE(orderby_1_Postfavorite.orderby_aggregator, $2) ASC
LIMIT $3 OFFSET $4
```

**Comment by nurul3101** [comment]
Hey @semurgx 👋 

It's not possible to change the generated SQL query, you can always tweak/optimise the generated SQL query and pass it to [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to execute it.

**Reply by semurgx** [reply]
@nurul31 Unfortunately, raw queries are not feasible for us. Are there any plans to support this feature in future releases?

**Reply by nurul3101** [reply]
How do you envision this to happen? I would recommend you create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that our engineering/product can provide their insights.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
